### PR TITLE
fix(#7825): imagery pause

### LINKED ISSUE #7825: [Imagery] Some images not being bounded properly
#### Summary
Thumbnail strip in imagery view is organized chronologically, ascending left to right, with most recent image telemetry on the right. If, paused, on resume, the paused image (and other older images) can get stuck to the right side of the thumbnail strip, if the strip was paused for too long. My guess is too long refers to when the images were to scroll off screen.

Upon further investigation, the root cause is images are not being removed if multiple images are to be removed in the same event. Pausing was the only way I could reliably reproduce this (although I'm not 100% sure why that plays a role).

VIPERGC-577

#### Expected vs Current Behavior
Expect on resume, all thumbnails are present and in order, and focus goes to the most recent, right most, image.

#### Steps to Reproduce

1. Navigate to an example imagery object, with the time conductor in realtime mode
2. Pause telemetry by clicking on an image, or clicking the pause button located above the thumbnail strip
3. Wait for a while... (My guess is until when the images were to scroll off screen)
4. Resume telemetry
5. Observe that when new telemetry comes in, the paused images are stuck to the right of the incoming telemetry (timestamp and staleness indicator will indicate)

#### Environment

* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [x] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available? Refresh or change time conductor bounds to force a reload
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by davetsay**
Closes #7825 

### Describe your changes:
Fixed `imageryData.dataRemoved` method to properly remove multiple images
Note: This is fixing the current implementation. A better approach would be to not rely on keeping a mirrored `imageHistory` in sync with the underlying `telemetryCollection`.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
Let's split out the test

**Comment by davetsay** [comment]
> Let's split out the test

this test runs against multiple imagery views

File: e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js [hunk 1]
```diff
@@ -746,6 +747,28 @@ async function performImageryViewOperationsAndAssert(page, layoutObject) {
   await page.clock.resume();
   await expect(page.getByLabel(lastImageTimestamp)).toBeHidden();
 
+  // go way forward in time to ensure multiple images are discarded
+  const IMAGES_TO_DISCARD_COUNT = 5;
```
Comment (ozyx) [inline e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js:751]: this is fantastic

**Review by ozyx — APPROVED** [review]
LGTM

File: e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js (hunk 1 above)
Comment (ozyx) [reply e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js:751]: (this is supposed to be for the whole test part)

Comment (davetsay) [reply e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js:751]: its okay. I know my `const`s are internationally famous

**Comment by davetsay** [comment]
@unlikelyzero , can you re-review? Do you still want me to break up the tests, or is this good as is?

File: e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js (hunk 1 above)
Comment (jvigliotta) [reply e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js:751]: Dave "Some of the Best Consts" Tsay

**Comment by davetsay** [comment]
@unlikelyzero , tests are updated and `@clock` tests are split out. Can you re-review?

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**